docs/api/utility-process.md

```diff
@@ -0,0 +1,85 @@
+# UtilityProcess
+
+`UtilityProcess` is the equivalent of [`child_process.fork`][] API from Node.js
+but instead uses [Services API][] from Chromium to create child processes with
+Node.js and Message ports enabled.
+
+`UtilityProcess` is an [EventEmitter][event-emitter].
+
+## Class: UtilityProcess
+
+> Child process with Node.js integration spawned by Chromium.
+
+Process: [Main](../glossary.md#main-process)<br />
+
+### `new UtilityProcess(modulePath[, args][, options])`
+
+* `modulePath` string - Absolute path to the script that should run as entrypoint in the child process.
+* `args` string[] (optional) - List of string arguments that will be available as `process.argv`
+  in the child process.
+* `options` Object (optional)
+  * `env` Object - Environment key-value pairs. Default is `process.env`.
+  * `execArgv` string[] - List of string arguments passed to the executable. Default is `process.execArgv`.
+  * `serviceName` string - Name of the process that will appear in `name` property of
+    [`child-process-gone` event of `app`](app.md#event-child-process-gone).
+    Default is `node.mojom.NodeService`.
+  * `disableLibraryValidation` boolean (optional) _macOS_ - Allows launching the process using
+    Helper (Plugin).app which can code signed with entitlement that disables library validation.
```

Comment:
By default generic child processes on macOS will be spawned with `Helper.app` executable, when applications have an extension architecture and the extensions can load native libraries (ex: node native modules), these libraries will not be codesigned by the application, the hardened runtime will prevent these unsigned libraries to be loaded into the process. Application has to codesign the `Helper.app` with the entitlements [com.apple.security.cs.disable-library-validation](https://developer.apple.com/documentation/bundleresources/entitlements/com_apple_security_cs_disable-library-validation), [com.apple.security.cs.allow-unsigned-executable-memory](https://developer.apple.com/documentation/bundleresources/entitlements/com_apple_security_cs_allow-unsigned-executable-memory) to support this use case.

But this causes the Application main process to respect `DYLD_INSERT_LIBRARIES` environment variable which serves as untrusted code injection vector https://knight.sc/malware/2019/03/15/code-injection-on-macos.html. Chrome also has [use case](https://source.chromium.org/chromium/chromium/src/+/main:content/browser/video_capture_service.cc;l=117-122) to load unsigned libraries in certain cases and they have reserved `Helper (Plugin).app` for this specific use case https://source.chromium.org/chromium/chromium/src/+/main:content/public/common/child_process_host.h;l=116-123.

I decided to take advantage of this support and with the above flag applications can force the utility process to be launched using the `Helper (Plugin).app` and can sign only this executable with the above privileged entitlements (ex: https://source.chromium.org/chromium/chromium/src/+/main:chrome/app/helper-plugin-entitlements.plist) leaving the main entry process unaffected. I was not sure if this would be majority use case hence didn't keep this as default. Also, I am not sure how to best explain this flag in the docs with short description, open to better documentation.

Reply:
Perhaps a better name would be `allowLoadingUnsignedLibraries`?

And document as:

> With this flag, the utility process will be launched via the `Electron Helper (Plugin).app` helper process on macOS, which by default is codesigned with the `com.apple.security.cs.disable-library-validation` and `com.apple.security.cs.allow-unsigned-executable-memory` entitlements. This will allow the utility process to load unsigned libraries. Unless you specifically need this capability, it's best to leave this disabled.